plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -39,114 +37,200 @@
+    private String vaultAppRole;
+    private String vaultAppRoleSecret;
+    private Calendar authTokenExpiration;
+
 
     public VaultSecretSource() {
+        authTokenExpiration = Calendar.getInstance();
         Optional<String> vaultFile = Optional.ofNullable(System.getenv(CASC_VAULT_FILE));
         Properties prop = new Properties();
         vaultFile.ifPresent(file -> readPropertiesFromVaultFile(file, prop));
 
         // Parse variables
-        Optional<String> vaultPw = getVariable(CASC_VAULT_PW, prop);
-        Optional<String> vaultUser = getVariable(CASC_VAULT_USER, prop);
-        Optional<String> vaultUrl = getVariable(CASC_VAULT_URL, prop);
-        Optional<String> vaultMount = getVariable(CASC_VAULT_MOUNT, prop);
-        Optional<String> vaultToken = getVariable(CASC_VAULT_TOKEN, prop);
-        Optional<String> vaultAppRole = getVariable(CASC_VAULT_APPROLE, prop);
-        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
-        Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
-        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
-
-        Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
+        Optional<String> vaultEngineVersionOpt = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
+        Optional<String> vaultUrlOpt = getVariable(CASC_VAULT_URL, prop);
+        Optional<String> vaultNamespaceOpt = getVariable(CASC_VAULT_NAMESPACE, prop);
+        Optional<String> vaultPwOpt = getVariable(CASC_VAULT_PW, prop);
+        Optional<String> vaultUserOpt = getVariable(CASC_VAULT_USER, prop);
+        Optional<String> vaultMountOpt = getVariable(CASC_VAULT_MOUNT, prop);
+        Optional<String> vaultTokenOpt = getVariable(CASC_VAULT_TOKEN, prop);
+        Optional<String> vaultAppRoleOpt = getVariable(CASC_VAULT_APPROLE, prop);
+        Optional<String> vaultAppRoleSecretOpt = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
+        Optional<String[]> vaultPathsOpt = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
                 .map(Optional::of)
                 .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
 
         // Check mandatory variables are set
-        if (!vaultUrl.isPresent() || !vaultPaths.isPresent()) return;
+        if (!vaultUrlOpt.isPresent() || !vaultPathsOpt.isPresent()) return;
 
         // Check defaults
-        if (!vaultMount.isPresent()) vaultMount = Optional.of(DEFAULT_USER_BACKEND);
-        if (!vaultEngineVersion.isPresent()) vaultEngineVersion = Optional.of(DEFAULT_ENGINE_VERSION);
+        vaultMount = vaultMountOpt.orElse(DEFAULT_USER_BACKEND);
+        String vaultEngineVersion = vaultEngineVersionOpt.orElse(DEFAULT_ENGINE_VERSION);
+
+        // Set class fields
+        String vaultUrl = vaultUrlOpt.get();
+        vaultPw = vaultPwOpt.orElse(null);
+        vaultUser = vaultUserOpt.orElse(null);
```

Comment:
all of these optional fields seem pretty pointless, they are all set to null here, I would remove optional or refactor the class design

Reply:
Restructured logic to place authentication logic outside